spec/models/credence_question_spec.rb

```diff
@@ -0,0 +1,92 @@
+require 'spec_helper'
+
+describe CredenceQuestion do
+  it 'should be able to create random questions' do
+    gen = create_valid_credence_question
+    as = (0..9).map do |rank|
+      create_valid_credence_answer(credence_question: gen, rank: rank)
+    end
+
+    q = gen.create_random_question
+    expect(q.class).to eq CredenceGameResponse
+  end
+
+  it 'should not create a question where both answers have the same rank' do
+    gen = create_valid_credence_question
+    [1, 1, 1, 1, 2].each do |rank|
+      create_valid_credence_answer(credence_question: gen, rank: rank)
+    end
+
+    100.times do
+      q = gen.create_random_question
+      expect(q.first_answer.rank).to_not eq q.second_answer.rank
+    end
+  end
+
+  it 'should uniformly distribute questions in aswer-space' do
+    pending "work out a good test to use"
+    raise "not yet implemented"
+
+    # gen.create_random_question is sufficiently slow that we don't want to do
+    # it loads of times. But if we don't do it enough, our test will be prone to
+    # failing randomly.
+    #   Is it possible to only have this test run if we request it explicitly?
```

Comment:
I see. That is an interesting scenario I have not encountered yet in testing.

I assume that the reason create_random_question is slow, is due to an active record being created every time the function is called. Here's what I would do. Separate the logic that generates the samples within the probability distribution into its own class like: SomeDistributionSampler with a .sample instance method. Then run the unit tests that checks the accuracy of the distribution on that class. If it's only simple arithmetic, you should be able to sample 10000+ before the tests start getting slow. Then use that distribution class in create_random_question and test that when you call create_random_question, SomeDistributionSampler gets a .sample call:

`expect_any_instance_of(SomeDistributionSampler).to receive(:sample).with('arguments', 'if', 'necessary')`